crates/subspace-farmer/src/single_disk_plot.rs

```diff
@@ -860,6 +889,58 @@ impl SingleDiskPlot {
                 }
             })?;
 
+        let (piece_reader, mut read_piece_receiver) = PieceReader::new();
```

Comment:
I was thinking about it. If you look at individual functions they are quire managable. Moving them out is certainly possible, but then we'll have a big number of arguments there.
I'm hoping with further consensus upgrades we'll have less entities in the system and probably ne able to shrink it. I'm inclined to not refactor it further until then. This is also why I preferred to copy-paste a few things again (like piece decoding) because it will be dropped anyway, so refactoring is mostly a waste of effort right now.

Reply:
I am thinking about it still. Those blocks are only `unsafe` because Rust can make no guarantees that files will not be modified. And those particular blocks are actually quite small, so just moving them out will IMO make code harder to read, not easier, you'll just have to jump around a lot and hide unsafe code behind thin safe interface that can't actually guarantee safety anyway.